### PR TITLE
Enforces project wide consistent code style using clang-format.

### PR DESCRIPTION
**Pull request by lexicalunit**

### CODE REVIEW

**Comment by mloskot** [comment]
:+1: 

A note on how to use clang-format for nanodbc would be useful too.

**Comment by lexicalunit** [comment]
@mloskot any thoughts on the format? I tried to pick a configuration for `.clang-format` that resulted in the least number of changes. Or to put it another way: A style that most matched the current style.

File: src/nanodbc.h [hunk 1]
```diff
@@ -398,9 +405,12 @@ class statement
...
         PARAM_IN //!< Binding an input parameter.
-        , PARAM_OUT //!< Binding an output parameter.
-        , PARAM_INOUT //!< Binding an input/output parameter.
-        , PARAM_RETURN //!< Binding a return parameter.
+        ,
+        PARAM_OUT //!< Binding an output parameter.
+        ,
+        PARAM_INOUT //!< Binding an input/output parameter.
+        ,
+        PARAM_RETURN //!< Binding a return parameter.
```
Comment (lexicalunit) [inline src/nanodbc.h]: @lexicalunit fix this

File: src/nanodbc.h [hunk 2]
```diff
@@ -491,15 +501,18 @@ class statement
     //! \param batch_operations Numbers of rows to fetch per rowset, or the number of batch parameters to process.
     //! \param timeout The number in seconds before query timeout. Default is 0 indicating no timeout.
     //! \return A result set object.
-    //! \attention You will want to use transactions if you are doing batch operations because it will prevent auto commits from occurring after each individual operation is executed.
+    //! \attention You will want to use transactions if you are doing batch operations because it will prevent auto
+    //!            commits from occurring after each individual operation is executed.
```
Comment (lexicalunit) [inline src/nanodbc.h]: @lexicalunit check that multi line comments like this are processed correctly by doxygen.

File: src/nanodbc.h [hunk 3]
```diff
@@ -895,7 +906,8 @@ class connection
 #if !defined(NANODBC_DISABLE_ASYNC)
     //! \brief Initiate an asynchronous connection operation to the given data source.
     //!
-    //! This method will only be available if nanodbc is built against ODBC headers and library that supports asynchronous mode.
+    //! This method will only be available if nanodbc is built against ODBC headers and library that supports
+    //! asynchronous mode.
```
Comment (lexicalunit) [inline src/nanodbc.h]: @lexicalunit fix comment here

File: src/nanodbc.h [hunk 4]
```diff
@@ -910,15 +922,16 @@ class connection
...
+        const string_type& user,
+        const string_type& pass,
+        void* event_handle,
+        long timeout = 0);
 
     //! \brief Initiate an asynchronous connection operation using the given connection string.
     //!
-    //! This method will only be available if nanodbc is built against ODBC headers and library that supports asynchronous mode.
+    //! This method will only be available if nanodbc is built against ODBC headers and library that supports
+    //! asynchronous mode.
```
Comment (lexicalunit) [inline src/nanodbc.h]: And here

**Comment by mloskot** [comment]
@lexicalunit Generally, I like it. I like the LLVM coding style and conventions.

I see several issues though:

1. `ColumnLimit:     120`  :-)
> We fall in love w/ high-res screens. We can actually now project the whole of the Serengeti ~Kevlin Henney in https://vimeo.com/97329157  @ 15:45
What do you think about sliding it it closer to 100 characters?

2.  What about using `BreakConstructorInitializersBeforeComma: true`?

3.  Did you decide to give up on the leading command in front of function arguments? No strong feelings, just a question.

4. I think some manual post-reformatting intervention will be required to bring places like below to better shape. Note, if `2147483647` read `2147483647 // SQL Server`, for instance, then that would be reformatted much nicer.
    ```
REQUIRE(
    (columns.column_size() == 2147483647 || columns.column_size() == 1048576 || // Vertica
     columns.column_size() == 65535 ||                                          // MySQL
     columns.column_size() ==
         8190 || // PostgreSQL uses MaxLongVarcharSize=8190, which is configurable in odbc.ini
     columns.column_size() == 0)); // SQLite
```

5. I think, one liners like this look less readable
    ``` 
NANODBC_CALL_RC(
    SQLSetStmtAttr, rc, stmt_, SQL_ATTR_PARAMSET_SIZE, (SQLPOINTER)(std::intptr_t)batch_operations, 0);
```
but, that might become motivation to switch to C++ casts, then clang-format will refactor to:
    ```
NANODBC_CALL_RC(
    SQLSetStmtAttr,
    rc,
    stmt_,
    SQL_ATTR_PARAMSET_SIZE,
    static_cast<SQLPOINTER>(static_cast<std::intptr_t>(batch_operations)),
    0);
```
Unless, you have decided the former is better.

In future, it is a good idea to keep it in mind how clang-format works, so we help it to format the code nicely.

p.s. I've just started using the clang-format extension for Visual Studio, awesome!

**Comment by lexicalunit** [comment]
1. I'm good with `ColumnLimit: 100` 👍 The default of 80 is too short for me though.
2. Ah I didn't see `BreakConstructorInitializersBeforeComma`. I'll enable that!
3. I didn't see a setting for that and I didn't have strong feelings on it either.
4. Yep, I'm adding line comments to places that need some manual work.
5. Yeah I looked for a way to make lines like that be broken into multiple lines. I think changing the column size to 100 instead of 120 will help.
6. Ah I didn't realize `///` was the preferred comment for Doxygen. I've just always used `//!`. I can change that too. It'll also mean changing `//!<` comments too. I'm not sure there's an equivalent way of expression that kind of notation when using the `///` style comments.

**Comment by mloskot** [comment]
Sounds great.

Re 6. [There is](https://www.stack.nl/~dimitri/doxygen/manual/docblocks.html#memberdoc) `///<` which seems to be the closest match, I think.

**Comment by mloskot** [comment]
LGTM!

We will master all other details as we go.

File: examples/rowset_iteration.cpp [hunk 5]
```diff
@@ -32,7 +32,8 @@ int main(int argc, char* argv[])
     try
     {
         // Example:
-        // "Driver={ODBC Driver 11 for SQL Server};Server=xxx.sqlserver.net;Database=mydb;UID=joe;PWD=secret;"
+        // "Driver={ODBC Driver 11 for SQL
```
Comment (lexicalunit) [inline examples/rowset_iteration.cpp]: Needs manual fixing.

File: examples/table_schema.cpp [hunk 6]
```diff
@@ -16,68 +16,108 @@
...
+        return "SQL_BIT";
+    case SQL_DATETIME:
+        return "SQL_DATETIME";
+    case SQL_TYPE_DATE:
+        return "SQL_TYPE_DATE";
+    case SQL_TYPE_TIME:
+        return "SQL_TYPE_TIME";
+    case SQL_TYPE_TIMESTAMP:
+        return "SQL_TYPE_TIMESTAMP";
+    // case SQL_TYPE_UTCDATETIME: return "SQL_TYPE_UTCDATETIME";
```
Comment (lexicalunit) [inline examples/table_schema.cpp]: Needs manual fixing.

File: examples/table_schema.cpp [hunk 7]
```diff
@@ -16,68 +16,108 @@
...
+    // case SQL_TYPE_UTCTIME:  return "SQL_TYPE_UTCTIME";
```
Comment (lexicalunit) [inline examples/table_schema.cpp]: Needs manual fixing.

File: examples/table_schema.cpp [hunk 8]
```diff
@@ -101,7 +141,8 @@ int main(int argc, char* argv[])
     try
     {
         // Example:
-        // "Driver={ODBC Driver 11 for SQL Server};Server=xxx.sqlserver.net;Database=mydb;UID=joe;PWD=secret;"
+        // "Driver={ODBC Driver 11 for SQL
```
Comment (lexicalunit) [inline examples/table_schema.cpp]: Needs manual fixing.

File: test/mssql_test.cpp [hunk 9]
```diff
@@ -9,15 +9,14 @@ namespace
 struct mssql_fixture : public base_test_fixture
 {
     mssql_fixture()
-        : base_test_fixture(/* connecting string from command line or NANODBC_TEST_CONNSTR environment variable)*/)
+        : base_test_fixture(
+              /* connecting string from command line or NANODBC_TEST_CONNSTR environment variable)*/)
```
Comment (lexicalunit) [inline test/mssql_test.cpp]: Needs manual fixing.

File: test/mysql_test.cpp [hunk 10]
```diff
@@ -6,19 +6,17 @@
 
 namespace
 {
-    struct mysql_fixture : public base_test_fixture
-    {
-        mysql_fixture()
+struct mysql_fixture : public base_test_fixture
+{
+    mysql_fixture()
```
Comment (lexicalunit) [inline test/mysql_test.cpp]: Needs manual fixing.

File: test/odbc_test.cpp [hunk 11]
```diff
@@ -6,19 +6,17 @@
 
 namespace
 {
-    struct odbc_fixture : public base_test_fixture
-    {
-        odbc_fixture()
+struct odbc_fixture : public base_test_fixture
+{
+    odbc_fixture()
         : base_test_fixture(/* connecting string from NANODBC_TEST_CONNSTR environment variable)*/)
```
Comment (lexicalunit) [inline test/odbc_test.cpp]: Needs manual fixing.

Comment (lexicalunit) [inline test/odbc_test.cpp]: Needs manual fixing.

File: test/postgresql_test.cpp [hunk 12]
```diff
@@ -6,19 +6,17 @@
 
 namespace
 {
-    struct postgresql_fixture : public base_test_fixture
-    {
-        postgresql_fixture()
+struct postgresql_fixture : public base_test_fixture
+{
+    postgresql_fixture()
         : base_test_fixture(/* connecting string from NANODBC_TEST_CONNSTR environment variable)*/)
```
Comment (lexicalunit) [inline test/postgresql_test.cpp]: Needs manual fixing.

File: test/vertica_test.cpp [hunk 13]
```diff
@@ -6,19 +6,17 @@
 
 namespace
 {
-    struct vertica_fixture : public base_test_fixture
-    {
-        vertica_fixture()
+struct vertica_fixture : public base_test_fixture
+{
+    vertica_fixture()
         : base_test_fixture(/* connecting string from NANODBC_TEST_CONNSTR environment variable)*/)
```
Comment (lexicalunit) [inline test/vertica_test.cpp]: Needs manual fixing.

**Pull request merged**